requests_pkcs12.py

```diff
@@ -197,40 +222,23 @@ def selftest():
         "withEncryption": {
             "pkcs12_password": b"correcthorsebatterystaple",
             "expected_status_code": 200,
+            "expected_exception_message": None,
         },
         "withEmptyPassword": {
             "pkcs12_password": b"",
             "expected_status_code": 200,
+            "expected_exception_message": "Password must be 1 or more bytes.",
         },
         "withoutEncryption": {
             "pkcs12_password": None,
             "expected_status_code": 200,
+            "expected_exception_message": None,
         },
     }
+
     for test_case_name, test_case in test_cases.items():
-        password = test_case['pkcs12_password']
         print(f"Testing {test_case_name}")
-        try:
-            algorithm = cryptography.hazmat.primitives.serialization.BestAvailableEncryption(password) \
-                if test_case['pkcs12_password'] is not None else cryptography.hazmat.primitives.serialization.NoEncryption()
-            pkcs12_data = cryptography.hazmat.primitives.serialization.pkcs12.serialize_key_and_certificates(
-                name=b'test',
-                key=key,
-                cert=cert,
-                cas=[cert, cert, cert],
-                encryption_algorithm=algorithm
-            )
-            response = get(
-                'https://example.com/',
-                pkcs12_data=pkcs12_data,
-                pkcs12_password=test_case['pkcs12_password']
-            )
-            if response.status_code != 200:
-                raise Exception('Unexpected response: {response!r}'.format(**locals()))
-        except ValueError as e:
-            if test_case == 'withEmptyPassword' and str(e) != 'Password must be 1 or more bytes.':
-                raise(e)
-            continue
+        execute_test_case(test_case, key, cert)
```

Comment:
Almost done. Now please get rid of the test_cases list and the loop, and just call that function multiple times.

Reply:
The same, please do a change, thank you!